File: tmp-smoke-meeting.js
fetch("http://127.0.0.1:3001/api/integra/cameras")
  .then((r) => r.json())
  .then(async (d) => {
    const cams = d.items || [];
    console.log("cams", cams.length);
    for (const c of cams) {
      const n = c.name || "";
      if (/meeting|ptz/i.test(n) || c.isPtz) {
        console.log(n, c.sourceIp, "ptz=" + c.isPtz, "audio=" + c.hasAudio);
      }
    }
    const ips = new Set(
      cams.filter((c) => /meeting/i.test(c.name || "")).map((c) => c.sourceIp).filter(Boolean),
    );
    ips.add("192.168.9.178");
    const ev = await (await fetch("http://127.0.0.1:3001/api/integra/push/events?sinceMs=180000&limit=80")).json();
    const items = ev.items || [];
    const hist = {};
    for (const e of items.filter((x) => x.eventType === "fielddetection")) {
      const n = (e.targets || []).length;
      hist[n] = (hist[n] || 0) + 1;
    }
    console.log("fd hist", hist);
    for (const e of items) {
      if (!ips.has(e.deviceIp)) continue;
      console.log(e.id, e.eventType, e.deviceIp, "n=" + (e.targets || []).length, JSON.stringify(e.targets || []).slice(0, 120));
    }
  })
  .catch((e) => {
    console.error(e);
    process.exit(1);
  });

File: tmp-smoke-meeting.py
#!/usr/bin/env python3
"""Probe Meeting Room fielddetection targets on prod API (localhost)."""
import json
import urllib.request

BASE = "http://127.0.0.1:4000"

def get(path: str):
    with urllib.request.urlopen(BASE + path, timeout=12) as r:
        return json.load(r)

cams = get("/integra/cameras").get("items") or []
meeting = [c for c in cams if "meeting" in (c.get("name") or "").lower()]
print("=== Meeting Room cams ===")
for c in meeting:
    print(c.get("name"), "ip=", c.get("sourceIp"), "hasAudio=", c.get("hasAudio"), "ptz=", c.get("isPtz"))

ips = {c.get("sourceIp") for c in meeting if c.get("sourceIp")}
if not ips:
    ips = {"192.168.9.178"}

events = get("/integra/push/events?sinceMs=120000&limit=80").get("items") or []
print("=== events last 120s for meeting ips ===")
for e in events:
    if e.get("deviceIp") not in ips:
        continue
    t = e.get("targets") or []
    print(
        e.get("id"),
        e.get("eventType"),
        e.get("deviceIp"),
        (e.get("occurredAt") or "")[:19],
        "nTargets=",
        len(t),
        t,
    )

print("=== target-count histogram (all fielddetection 120s) ===")
from collections import Counter
hist = Counter()
for e in events:
    if e.get("eventType") != "fielddetection":
        continue
    hist[len(e.get("targets") or [])] += 1
print(dict(hist))
